### ISSUE TITLE
resource "aws_lb" subnet mapping private ip

### ISSUE THREAD

**Issue by bailey84j** [terraform-providers/terraform-provider-aws#11674]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Due to Firewall requirements I need to assign private ips to the NLB which I can do via the cli and UI but cannot do via terraform

### New or Affected Resource(s)

 aws_lb

### Potential Terraform Configuration

```hcl
resource "aws_lb" "example" {
  name               = "example"
  load_balancer_type = "network"

  subnet_mapping {
    subnet_id     = "${aws_subnet.example1.id}"
    private_ip = "${aws_network_interface.example1_ni.id}"
  }

  subnet_mapping {
    subnet_id     = "${aws_subnet.example2.id}"
    private_ip = "${aws_network_interface.example2_ni.id}"
  }
}
```

**Comment by ewbankkit** [comment]
I think this is the same use case as https://github.com/terraform-providers/terraform-provider-aws/issues/11044.

**Comment by bflad** [comment]
Hi folks 👋  This functionality has been merged and will release with version 3.1.0 of the Terraform AWS Provider, next Thursday. Thanks to @KadenLNelson for the implementation. 👍
